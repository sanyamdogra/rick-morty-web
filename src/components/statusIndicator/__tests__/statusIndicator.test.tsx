import React from "react";
import { render, screen, cleanup } from "@testing-library/react";
import StatusIndicator from "..";

afterEach(cleanup);

test("should match the snapshot and render correct data", () => {
  const { asFragment } = render(<StatusIndicator status="Alive" />);

  expect(asFragment()).toMatchSnapshot();

  const indicator = screen.getByTestId("status-indicator");

  expect(
    indicator.classList.contains("rick-morty-status-indicator--alive")
  ).toBe(true);

  expect(indicator).toHaveAttribute("title", "Alive");
});
