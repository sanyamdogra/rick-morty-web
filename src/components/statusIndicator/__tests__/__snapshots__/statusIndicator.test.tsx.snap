// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot and render correct data 1`] = `
<DocumentFragment>
  <div
    class="rick-morty-status-indicator rick-morty-status-indicator--alive"
    data-testid="status-indicator"
    title="Alive"
  />
</DocumentFragment>
`;
